corehq/ex-submodules/casexml/apps/case/dbaccessors/related.py

```diff
@@ -58,11 +61,15 @@ def _row_to_index_info(row):
             relationship=row['value']['relationship']
         )
 
-    return list(map(_row_to_index_info, CommCareCase.get_db().view(
-        'case_indices/related',
-        keys=_get_keys_for_reverse_index_view(domain, case_ids, relationship),
-        reduce=False,
-    )))
+    return [
+        _row_to_index_info(row)
+        for row in CommCareCase.get_db().view(
+            'case_indices/related',
+            keys=_get_keys_for_reverse_index_view(domain, case_ids, relationship),
+            reduce=False,
+        )
+        if not exclude_for_case_type or row['value']['referenced_type'] != exclude_for_case_type
```

Comment:
Perhaps `NO_MATCH` magic is also not any simpler to wrap one's head around while reading the code. But sure.

Reply:
Unfortunately the change is not quite as simple as I implied here because of https://github.com/dimagi/commcare-hq/pull/28998#discussion_r559624751. I think the default would need to be changed in each method that accepts the `exclude_for_case_type` argument. However, this seems less than ideal since it's a non-standard default value, given how many places it will need to be referenced. I think the choice is either revert back to the way it was before or use `exclude_for_case_type=NO_MATCH` in each method with that argument. You can decide which you prefer.

Reply:
Ah, true. I will revert back to the way it was. Thanks for the detailed note